channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/dto/JaicpNativeBotRequest.kt

```diff
@@ -22,6 +24,25 @@ interface TelephonyBotRequest : JaicpNativeBotRequest {
         get() = jaicp.rawRequest["extension"]?.jsonPrimitive?.content
     val calleePayload: JsonObject?
         get() = jaicp.rawRequest.jsonObject["originateData"]?.jsonObject?.get("payload")?.jsonObject
+    val campaignSchedule: JsonObject?
+        get() = jaicp.rawRequest.jsonObject["originateData"]?.jsonObject?.get("callScenarioData")
+            ?.jsonObject?.get("schedule")?.jsonObject?.get("campaignSchedule")?.jsonObject
+    val dialSchedule: JsonObject?
+        get() = jaicp.rawRequest.jsonObject["originateData"]?.jsonObject?.get("callScenarioData")
+            ?.jsonObject?.get("schedule")?.jsonObject?.get("dialSchedule")?.jsonObject
+    val audioToken: String?
+        get() = jaicp.data?.jsonObject?.get("resterisk")?.jsonObject?.get("callRecordsDownloadData")
+            ?.jsonObject?.get("audioToken")?.jsonObject?.get("token")?.jsonPrimitive?.content
+    val callRecordingPath: String?
+        get() = jaicp.data?.jsonObject?.get("resterisk")?.jsonObject?.get("callRecordingFile")?.jsonPrimitive?.content
+
+    fun getCallRecordingFullUrl(context: ActionContext<*, *, *>): String? =
```

Comment:
Why not
`val DefaultActionContext.callRecordingFullUrl: String?`

or even
`val ActionContext<out ActivatorContext, TelephonyBotRequest, TelephonyReactions, *>.callRecordingFullUrl: String?`